src/Algebra/Graph/Algorithm/Internal.hs

```diff
@@ -1,30 +1,38 @@
 module Algebra.Graph.Algorithm.Internal where
 
-import           Algebra.Graph
+import                            Algebra.Graph
+import Data.DisjointSet           (DisjointSet)
```

Comment:
Is there any reason to import `DisjointSet` specifically instead of using `DS.DisjointSet`?

Reply:
Remove `DS.` prefix in type declarations:
`components :: Ord a => Graph a -> DisjointSet a`

Reply:
This is common import practice, similar to

```haskell
import Data.IntMap (IntMap)
import qualified Data.IntMap as IntMap
```

This is recommended in the [documentation of `containers`](https://hackage.haskell.org/package/containers-0.6.2.1/docs/Data-IntMap.html), for instance.

One comment would be that `DS` is not worth abbreviation, so I would go with

```haskell
import Data.DisjointSet (DisjointSet)
import qualified Data.DisjointSet as DisjointSet
```